scripts/zephyr_module.py

```diff
@@ -33,6 +33,9 @@
 # the build system.
 type: map
 mapping:
+  name:
+    required: false
```

Comment:
if it is 'shall', shouldn't this be required?

Reply:
it's a shall for modules maintained by Zephyr community to be enforced through reviews, but what users decide downstream is their responsibility.

Also, enforcing this right away would require all modules to be updated, so keeping it optional ensures backward compatibility.

This could be changed to `required: true` in later release.